### ISSUE TITLE
Package installations fail intermittently on SLES15 and openSUSE Leap 15.x

### ISSUE THREAD

**Issue by johanneskastl** [ansible/ansible#59686]
##### SUMMARY
Package installations fail intermittently on SLES15 and openSUSE Leap 15.x

Apparently (I am not a python expert) python3-xml is no longer a separate package but is included in python3-base on those releases, maybe this might be related?

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/zypper.py

Probably [this line
](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/zypper.py#L197)

##### ANSIBLE VERSION
Tested with ansible 2.8.1, 2.8.2 and 2.8.3 installed via RPM package and 2.8.2 installed via virtualenv on multiple controller hosts (openSUSE Tumbleweed, SLES15, openSUSE Leap 15.x), here are some examples

```
ansible 2.8.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/foo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.3 (default, Apr 09 2019, 05:18:21) [GCC]
```
```
ansible 2.8.2
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/foo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.5 (default, Mar 31 2018, 19:45:04) [GCC]
```

```
ansible 2.8.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/foo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.5 (default, Mar 31 2018, 19:45:04) [GCC]
```
```
ansible 2.8.2
  config file = /home/foo/bar/ansible/ansible.cfg
  configured module search path = ['/home/foo.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/foo/.virtualenv-20190728/lib/python3.7/site-packages/ansible
  executable location = /home/foo/virtualenv-20190728/bin/ansible
  python version = 3.7.3 (default, Apr 09 2019, 05:18:21) [GCC]
```

##### CONFIGURATION

```
(empty)
```

##### OS / ENVIRONMENT
Tested with ansible 2.8.1, 2.8.2 and 2.8.3 installed via RPM package and 2.8.2 installed via virtualenv on multiple controller hosts (openSUSE Tumbleweed, SLES15, openSUSE Leap 15.x)

##### STEPS TO REPRODUCE

```
    - name: install chrony
      package:
        name: chrony
        state: present
      register: result
      until: result is succeeded
      when: ntp_daemon_type == "chronyd"

```
or

```
  - name: "openSUSE 15.x or SLES: upgrade all packages"
    zypper:
      name: '*'
      state: latest
    notify:
      - Reboot
      - "Wait for the machine(s) to be ready again"
    when: ansible_distribution == "SLES" or ansible_distribution == "openSUSE Leap"
```

##### EXPECTED RESULTS
The packages should be installed or upgraded.

##### ACTUAL RESULTS

```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ImportError: No module named xml
fatal: [some-hostname]: FAILED! => changed=false 
  attempts: 3
  module_stderr: |-
    Traceback (most recent call last):
      File "<stdin>", line 114, in <module>
      File "<stdin>", line 106, in _ansiballz_main
      File "<stdin>", line 49, in invoke_module
      File "/tmp/ansible_zypper_payload_MEqOfa/__main__.py", line 195, in <module>
    ImportError: No module named xml
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1
```

```
fatal: [some-other-hostname]: FAILED! => {"changed": false, "module_stderr": "Warning: Permanently added 'W.X.Y.Z' (ECDSA) to the list of known hosts.\r\nConnection to W.X.Y.Z closed.\r\n", "module_stdout": "Traceback (most recent call last):\r\n  File \"/home/tux/.ansible/tmp/ansible-tmp-1564256757.9409902-200923068294000/AnsiballZ_zypper_repository.py\", line 114, in <module>\r\n    _ansiballz_main()\r\n  File \"/home/tux/.ansible/tmp/ansible-tmp-1564256757.9409902-200923068294000/AnsiballZ_zypper_repository.py\", line 106, in _ansiballz_main\r\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\r\n  File \"/home/tux/.ansible/tmp/ansible-tmp-1564256757.9409902-200923068294000/AnsiballZ_zypper_repository.py\", line 49, in invoke_module\r\n    imp.load_module('__main__', mod, module, MOD_DESC)\r\n  File \"/tmp/ansible_zypper_repository_payload_9ayxwK/__main__.py\", line 397, in <module>\r\n  File \"/tmp/ansible_zypper_repository_payload_9ayxwK/__main__.py\", line 370, in main\r\n  File \"/tmp/ansible_zypper_repository_payload_9ayxwK/__main__.py\", line 198, in repo_exists\r\n  File \"/tmp/ansible_zypper_repository_payload_9ayxwK/__main__.py\", line 152, in _parse_repos\r\nImportError: No module named xml.dom.minidom\r\n", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}

```

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/cloud/linode/linode.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/linode/linode.py)
* [`lib/ansible/modules/packaging/os/zypper.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/zypper.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @AnderEnder @InTheCloudDan @alxgu @andytom @displague @lwm @rmcintosh @zbal
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @AnderEnder @InTheCloudDan @alxgu @andytom @displague @lwm @rmcintosh @zbal
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by netzvieh** [comment]
!component lib/ansible/modules/cloud/linode/linode.py

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/packaging/os/zypper.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/zypper.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by johanneskastl** [comment]
I did some more debugging, and found that this appears on both Leap 15.0 and Leap 15.1, but not reproducible.

One day creating new VMs from the same image everything works, next day it fails.

The one task most likely to end in this error is this:
```
  - name: "openSUSE 15.x or SLES: upgrade all packages"
    zypper:
      name: '*'
      state: latest
    notify:
      - Reboot
      - "Wait for the machine(s) to be ready again"
    when: ansible_distribution == "SLES" or ansible_distribution == "openSUSE Leap"
```

**Comment by alxgu** [comment]
Seems to be an issue with your environment/image. On my SLES15 server, it works as expected.

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/modules/packaging/os/zypper.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/packaging/os/zypper.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by losuler** [comment]
I came across the same error whilst using the JeOS image of OpenSUSE 15.1. However I discovered that it was caused by the installation of `fail2ban` (where Python 2.7 is a dependency as `python-base`). 

The installation of Python 2.7 causes it to become the default interpreter, which I assume the `xml` library isn't included in (like it is in `python3-base`), which is needed for the `zypper` module. In which the use of the `ansible_python_interpreter` variable would be a solution.

Thought I might mention it in case this is the same reason it's occuring for @johanneskastl, or in the case it's not, for anyone else who mistakenly think they have the same issue.
